### PR TITLE
[#IP-271]  Add Upsert Service Preferences endpoint

### PR DESCRIPTION
**Pull request by gquadrati**
#### List of Changes

- Add `UpsertServicePreference` endpoint
- no caching mangement 

#### Motivation and Context

Move to Opt-in model for services

#### How Has This Been Tested?

Unit tests and io-mock

#### Types of changes

- [] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/controllers/servicesController.ts [hunk 1]
```diff
@@ -34,6 +39,65 @@ export default class ServicesController {
...
+    | IResponseErrorConflict
+    | IResponseErrorTooManyRequests
+    | IResponseSuccessJson<ServicePreference>
+  > =>
+    withUserFromRequest(req, async user =>
+      withValidatedOrValidationError(
+        ServiceId.decode(req.params.id),
+        serviceId =>
+          withValidatedOrValidationError(
+            ServicePreference.decode(req.body.body),
```
Comment (BurnedMarshal) [inline src/controllers/servicesController.ts:84]: Why a double body? This is the body from the express request.
```suggestion
            ServicePreference.decode(req.body),
```

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: src/controllers/servicesController.ts (hunk 1 above)
Comment (gquadrati) [reply src/controllers/servicesController.ts:84]: Right, sorry..

**Review by BurnedMarshal — APPROVED** [review]
LGTM

**Pull request merged**